### PR TITLE
feat: split plugin server healthcheck into readiness and health

### PR DESCRIPTION
**Pull request by yakkomajuri**
ECS only supported healthchecks, but now that both self-hosted and Cloud are k8s, we can use readiness and health checks.

Need to update the chart as well

### CODE REVIEW

**Comment by yakkomajuri** [comment]
splut :D

File: plugin-server/src/main/services/http-server.ts [hunk 1]
```diff
@@ -50,6 +44,24 @@ export function createHttpServer(hub: Hub, serverInstance: ServerInstance, serve
                 res.statusCode = 503
                 res.end(JSON.stringify(responseBody))
             }
+        } else if (req.url === '/_ready' && req.method === 'GET') {
```
Comment (hazzadous) [inline plugin-server/src/main/services/http-server.ts:47]: What's the difference between `_health` and `_ready` and when do we use one or the other? Typically we have a livenessProbe which is just the verify that the container is completely dead and a readinessProbe which will for instance let us know when we can consider decommissioning the other pods in a deployment.

Maybe will be clearer with the Helm Chart changes also.

File: plugin-server/src/main/services/http-server.ts [hunk 2]
```diff
@@ -50,6 +44,24 @@ export function createHttpServer(hub: Hub, serverInstance: ServerInstance, serve
...
+            // Check that, if the server should have a kafka queue,
+            // the Kafka consumer is ready to consume messages
+            if (!serverInstance.queue || serverInstance.queue.consumerReady) {
```
Comment (hazzadous) [inline plugin-server/src/main/services/http-server.ts:50]: Is there a point where we haven't set `serverInstance.queue` but will eventually? I'm thinking if there is a state where we'd mark it as ready but it isn't yet.

File: plugin-server/src/main/services/http-server.ts (hunk 1 above)
Comment (yakkomajuri) [reply plugin-server/src/main/services/http-server.ts:47]: `_health` = livenessProbe, e2e check that kafka works
`_ready` = readinessProbe, checks that the Kafka consumer initially joined the group

File: plugin-server/src/main/services/http-server.ts (hunk 2 above)
Comment (yakkomajuri) [reply plugin-server/src/main/services/http-server.ts:50]: Nope, you'll see in `pluginsServer.ts` that the queue is always set up before the server

File: plugin-server/src/main/services/http-server.ts (hunk 1 above)
Comment (hazzadous) [reply plugin-server/src/main/services/http-server.ts:47]: Ok. Liveness is going to trigger a restart of the container, which we normally do if we know the container is dead and will not recover. For instance, the process is completely deadlocked.

I'm not really sure what the best thing to do for a Kafka consumer is however. The risk is we end up killing containers as they are in the process of rebalancing etc i.e. actively recovering but I don't know enough to be concrete.

**Review by hazzadous — APPROVED** [review]

**Pull request merged**